### PR TITLE
GNOME: 40 → 41

### PR DESCRIPTION
**Pull request by jtojnar**
https://discourse.gnome.org/t/gnome-41-released/7603/1

cc @NixOS/gnome 

https://github.com/NixOS/nixpkgs/projects/41
https://hydra.nixos.org/jobset/nixpkgs/gnome

### CODE REVIEW

File: pkgs/development/libraries/gtk/4.x.nix [hunk 1]
```diff
@@ -37,6 +33,7 @@
 , x11Support ? stdenv.isLinux
 , waylandSupport ? stdenv.isLinux
 , libGL
+, vulkanSupport ? false
```
Comment (mweinelt) [inline pkgs/development/libraries/gtk/4.x.nix:36]: Why is `vulkanSupport` disabled by default, when it was previously enable when not on Darwin?

Comment (bobby285271) [reply pkgs/development/libraries/gtk/4.x.nix:36]: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3391/diffs#ca9bb7eff80503c97c83505e8acea4002fd87ac6_58_54 it is experimental (and can cause crashes in inspector, according to the commit message from jtojnar).

Comment (mweinelt) [reply pkgs/development/libraries/gtk/4.x.nix:36]: Worth a comment I guess.

**Comment by maxeaubrey** [comment]
`libvirt-glib` doesn't build with glib > 2.69 on the latest release. Reading through commits doesn't look like it's been addressed yet, and after a quick search through the mailing list I didn't see any patches either (not that I was super thorough).

**Comment by bobby285271** [comment]
Some probably off-topic note: We just pinned Vala to 0.52 for Pantheon packages that cannot build with 0.54 [in this PR](https://github.com/NixOS/nixpkgs/pull/139163).

> `libvirt-glib` doesn't build with glib > 2.69 on the latest release.

A newbie question: I tried building `libvirt-glib` on this PR and it failed with `G_TYPE_IS_FINAL` not available before 2.70 ([full log](https://gist.github.com/bobby285271/0b896ace8b941dce5b29180227108c2b)). Will `mesonFlags = [ "-Dc_args=-DGLIB_DISABLE_DEPRECATION_WARNINGS" ];` *work* in this case (it does successfully *compile* for me)?

Not sure if these are related: (1) [`GLIB_VERSION_MAX_ALLOWED` set in `libvirt-glib`](https://github.com/libvirt/libvirt-glib/blob/9a34c4ea55e0246c34896e48b8ecd637bc559ac7/meson.build#L521) (2) [Fedora Rawhide is building this with `glib` 2.69.2 +  `gobject-introspection` 1.68.0](https://kojipkgs.fedoraproject.org//packages/libvirt-glib/4.0.0/3.fc36/data/logs/aarch64/build.log) (3) [Add "final" class attribute merge request in `gobject-introspection`](https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/257) (4) [Add "final" type flag merge request in `glib`](https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1937).

File: pkgs/desktops/gnome/core/gnome-desktop/bubblewrap-paths.patch [hunk 2]
```diff
@@ -1,15 +1,17 @@
+diff --git a/libgnome-desktop/gnome-desktop-thumbnail-script.c b/libgnome-desktop/gnome-desktop-thumbnail-script.c
+index ddcc1511..546c2a36 100644
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome/core/gnome-desktop/bubblewrap-paths.patch:2]:
```suggestion
```

File: pkgs/development/libraries/gtk/4.x.nix [hunk 3]
```diff
@@ -86,27 +83,25 @@ stdenv.mkDerivation rec {
     python3
     sassc
     gi-docgen
+    libxml2 # for xmllint
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtk/4.x.nix:86]: Should we make an extra output for xmllint?

File: pkgs/development/libraries/libsoup/default.nix [hunk 4]
```diff
@@ -23,21 +23,13 @@
...
+  version = "2.74.0";
 
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "11skbyw2pw32178q3h8pi7xqa41b2x4k6q4k9f75zxmh8s23y30p";
+    sha256 = "sha256-M7HU4NY5RWxnXCJ4d+lKgHjXMSM+LVdonBGrzvfTxI4=";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsoup/default.nix:34]: Please remove the unused fetchpatch from inputs.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -17541,6 +17541,8 @@ with pkgs;
 
   libsoup = callPackage ../development/libraries/libsoup { };
 
+  libsoup3 = callPackage ../development/libraries/libsoup/3.x.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:17544]:
```suggestion
  libsoup_3 = callPackage ../development/libraries/libsoup/3.nix { };
```

File: pkgs/development/libraries/libsoup/3.x.nix [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    "-Dpkcs11_tests=disabled"
+  ] ++ lib.optionals (!stdenv.isLinux) [
+    "-Dsysprof=disabled"
+  ];
+
+  # TODO: For some reason the pkg-config setup hook does not pick this up.
+  PKG_CONFIG_PATH = "${libnghttp2.dev}/lib/pkgconfig";
+
+  # HSTS tests fail.
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsoup/3.x.nix:79]: Can we disable only this test?

File: pkgs/development/libraries/libsoup/3.x.nix [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+, ninja
+, pkg-config
+, gnome
+, libsysprof-capture
+, sqlite
+, glib-networking
+, gobject-introspection
+, withIntrospection ? stdenv.buildPlatform == stdenv.hostPlatform
+, vala
+, withVala ? stdenv.buildPlatform == stdenv.hostPlatform
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsoup/3.x.nix:15]: Shouldn't this be one input called crossCompile or similar?

File: pkgs/development/libraries/libsoup/3.x.nix (hunk 6 above)
Comment (jtojnar) [reply pkgs/development/libraries/libsoup/3.x.nix:79]: Not without patching those test declarations out. https://github.com/mesonbuild/meson/issues/6999

File: pkgs/development/libraries/libsoup/3.x.nix (hunk 7 above)
Comment (jtojnar) [reply pkgs/development/libraries/libsoup/3.x.nix:15]: Previously, `stdenv.isCross` was used but the cross people switched to this.

File: pkgs/development/libraries/gtk/4.x.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/development/libraries/gtk/4.x.nix:86]: Unfortunately, that currently is not really possible to do conveniently – build systems generally support only single `bindir` per project. (Eventually, I want to have Debian-level granularity packages but that would require massive redesign of Meson and porting the affected projects to Meson.) Though, I wonder if we should explicitly list the outputs used – it used to be that way in the past.

**Comment by jtojnar** [comment]
All packages under `gnome` attribute have been bumped and possible problems noticed in changelogs were added to the [board](https://github.com/NixOS/nixpkgs/projects/41). Probably not ready to test on real systems yet. Feel free to lend a hand in triaging them or submitting fixes. Helping with red jobs on [Hydra](https://hydra.nixos.org/jobset/nixpkgs/gnome) also appreciated.

File: pkgs/desktops/gnome/core/gnome-terminal/default.nix [hunk 8]
```diff
@@ -56,11 +62,15 @@ stdenv.mkDerivation rec {
     nautilus # For extension
   ];
 
-  # Silly ./configure, it looks for dbus file from gnome-shell in the
+  # Silly build system, it looks for dbus file from gnome-shell in the
   # installation tree of the package it is configuring.
   postPatch = ''
-    substituteInPlace configure --replace '$(eval echo $(eval echo $(eval echo ''${dbusinterfacedir})))/org.gnome.ShellSearchProvider2.xml' "${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml"
-    substituteInPlace src/Makefile.in --replace '$(dbusinterfacedir)/org.gnome.ShellSearchProvider2.xml' "${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml"
+    substituteInPlace src/meson.build --replace "gt_prefix / gt_dbusinterfacedir / 'org.gnome.ShellSearchProvider2.xml'" "'${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml'"
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome/core/gnome-terminal/default.nix:68]:
```suggestion
    substituteInPlace src/meson.build \
      --replace "gt_prefix / gt_dbusinterfacedir / 'org.gnome.ShellSearchProvider2.xml'" \
      "'${gnome.gnome-shell}/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml'"
```

File: pkgs/development/libraries/nv-codec-headers/11_x.nix [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "5d6LCKQB31UZ0veanSeKJVrPkJ8o2nvQWRfIG8YuekM=";
+  };
+
+  makeFlags = [
+    "PREFIX=$(out)"
+  ];
+
+  meta = with lib; {
+    description = "FFmpeg version of headers for NVENC";
+    homepage = "https://ffmpeg.org/";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nv-codec-headers/11_x.nix:22]:
```suggestion
    homepage = "https://git.videolan.org/?p=ffmpeg/nv-codec-headers.git";
```

File: pkgs/development/libraries/nv-codec-headers/11_x.nix [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    license = licenses.mit;
+    maintainers = with maintainers; [
+      MP2E
+    ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nv-codec-headers/11_x.nix:26]:
```suggestion
    maintainers = with maintainers; [ MP2E ];
```

File: pkgs/development/libraries/atkmm/2.36.nix [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+  passthru = {
+    updateScript = gnome.updateScript {
+      packageName = pname;
+      versionPolicy = "odd-unstable";
+    };
+  };
+
+  meta = {
+    description = "C++ wrappers for ATK accessibility toolkit";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/atkmm/2.36.nix:28]: Missing maintainer

**Comment by t184256** [comment]
Yeah, and probably makes sense to update gnome extensions in the same PR.

**Comment by jtojnar** [comment]
Weird, that does not look like something that should fail:

https://gitlab.gnome.org/GNOME/tracker/-/blob/255c935754ca6c0b6a0e05fd411c21c7364ea5cd/tests/libtracker-sparql/tracker-resource-test.c#L61

I was not able to reproduce it so probably something transient.

**Comment by jtojnar** [comment]
All GNOME apps seem to work, the portal too, failing Hydra packages are pretty low (only few leaf Vala packages) so the only remaining thing is extensions but those can be updated separately when they are released.

Unfortunately, it turns out the tests no longer pass. Likely because the shell api we used is now private: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1970

**Comment by fabianhjr** [comment]
Shouldn't this merge request target staging instead of the main branch?

Thanks for working on this. Enjoying GNOME 41 so far and nothing seems to be broken after a whole day of usage(other than some extensions missing upstream GNOME 41 support unu)

**Comment by jtojnar** [comment]
Since we build the packages on Hydra, merging to master will not cause a mass rebuild.

**Comment by t184256** [comment]
I somehow still can't figure out a way to rebase this to master and not trigger a huge rebuild, so I'm mildly cautious about that claim.

Running it for 3 days already, still can't find much to complain about
(which kinda undermines my whole reason for trying it: dispelling boredom)
Brightness icon is occasionally volume icon and vice versa when I adjust them, but I seriously doubt this is a packaging issue.

Got any testing guidelines or something?

**Comment by TredwellGit** [comment]
`pkgsi686Linux.tracker` fails same as https://github.com/NixOS/nixpkgs/pull/139144#issuecomment-937302301.

**Comment by maxhille** [comment]
`nixos-unstable` is now broken for me, too.

```
running tests
ninja: Entering directory `/build/tracker-3.2.0/build'
ninja: no work to do.
 1/33 tracker:functional / fts-functions                      OK               0.89s
 2/33 tracker:functional / sparql-bugs                        OK               0.92s
 3/33 tracker:functional / query                              OK               1.32s
 4/33 tracker:functional / group-concat                       OK               0.68s
 5/33 tracker:functional / coalesce                           OK               0.77s
 6/33 tracker:functional / insertion                          OK               1.84s
 7/33 tracker:functional / distance                           OK               0.73s
 8/33 tracker:functional / graph                              OK               0.92s
 9/33 tracker:functional / collation                          OK               0.70s
10/33 tracker:functional / notifier                           OK               1.02s
11/33 tracker:functional / concurrent-query                   OK               1.49s
12/33 tracker:functional / ontology-rollback                  OK               0.47s
13/33 tracker:functional+ipc / bus-query-cancellation         OK               0.40s
14/33 tracker:gvdb / gvdb                                     OK               0.01s
15/33 tracker:common / file-utils                             OK               0.01s
16/33 tracker:common / utils                                  OK               0.01s
17/33 tracker:data / crc32                                    OK               0.02s
18/33 tracker:data / ontology-change                          OK               0.58s
19/33 tracker:data / ontology-error                           OK               0.03s
20/33 tracker:data / sparql-blank                             OK               0.07s
21/33 tracker:functional / cli                                OK               1.74s
22/33 tracker:data / insert-or-replace                        OK               0.37s
23/33 tracker:functional / ontology-changes                   OK               2.60s
24/33 tracker:functional / portal                             OK               2.89s
25/33 tracker:fts / fts                                       OK               0.39s
26/33 tracker:resource / resource                             FAIL             0.12s   killed by signal 6 SIGABRT
>>> MALLOC_PERTURB_=186 LANG=en_US.UTF8 TRACKER_LANGUAGE_STOP_WORDS_DIR=/build/tracker-3.2.0/src/libtracker-common/stop-words /build/tracker-3.2.0/build/tests/libtracker-sparql/tracker-resource-test
 ✀
stdout:
# random seed: R02S9bb64c7f3a28fd1d4cbc9738781ee4e8
# Testing Tracker resource abstraction
1..6
# Start of libtracker-sparql tests
# Start of tracker-resource tests
ok 1 /libtracker-sparql/tracker-resource/get_empty
Bail out! Tracker:ERROR:../tests/libtracker-sparql/tracker-resource-test.c:61:test_resource_get_set_simple: 'tracker_resource_get_first_double (resource, "http://example.com/1") == 0.6' should be TRUE
stderr:
**
Tracker:ERROR:../tests/libtracker-sparql/tracker-resource-test.c:61:test_resource_get_set_simple: 'tracker_resource_get_first_double (resource, "http://example.com/1") == 0.6' should be TRUE


27/33 tracker:data+slow / ontology                            OK               3.64s
28/33 tracker:data / service                                  OK               5.40s
29/33 tracker:data+slow / sparql                              OK               6.75s
30/33 tracker:sparql / sparql                                 OK               0.44s
31/33 tracker:sparql / statement                              OK               0.57s
32/33 tracker:examples / endpoint                             OK               0.59s
33/33 tracker:sparql / fd                                     OK               0.91s

Summary of Failures:

26/33 tracker:resource / resource                     FAIL             0.12s   killed by signal 6 SIGABRT


Ok:                 32
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0

Full log written to /build/tracker-3.2.0/build/meson-logs/testlog.txt
builder for '/nix/store/d67j41d88js80c6rvk7d6fpap7ilmczd-tracker-3.2.0.drv' failed with exit code 1
cannot build derivation '/nix/store/yw942n07506d9gfkcxc2j0mz83i15j39-gtk+3-3.24.30.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/vh65pnvg2h1rv5gcm6kbrk0y4cwbvv77-steam-usr-multi.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xv3xrn6fs6w2pyds8dh9hvwcp8vs2sg2-steam-fhs.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1ivxvf5svrz2k27bgcrz0a5k1hfiwm3d-steam.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2g1vib9a53hlx3vn5qkgnlpm8c9mpvp5-steam.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/5qyk5w68qbq5cl0adgp05pzwr0ynxgh7-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r911pdavjdg67sfpi19cqkab2nkc34qp-nixos-system-venus-21.11.git.7366ad6608f.drv': 1 dependencies couldn't be built
error: build of '/nix/store/r911pdavjdg67sfpi19cqkab2nkc34qp-nixos-system-venus-21.11.git.7366ad6608f.drv' failed
```

**Pull request merged**